test/metabase/test/data/interface.clj

```diff
@@ -61,6 +61,7 @@
     ;; default is nullable
    [:not-null?         {:optional true} [:maybe :boolean]]
    [:unique?           {:optional true} [:maybe :boolean]]
+   [:pk?               {:optional true} [:maybe :boolean]]
```

Comment:
I disagree with this design, primary keys should be a property of a table and not a property of a field. One practical limitation of this approach is that it cannot support ordered composite primary keys.

That's why for uploads, I changed the design so that the primary key is defined outside of the `column-definitions` of the `create-table!` args
See https://github.com/metabase/metabase/blob/3f351e4ec4b72060743f62277b980646efc58d4a/src/metabase/driver.clj#L919-L925

Reply:
yea, but I'm not sure about the practicality of that. Do you think we need to create composite PKs that often?

Reply:
Not often, but when we do, we'll either need to make the design more complicated by allowing two places to define primary keys. It would be nicer if it were in one place. I'm just wary that we'll want to change this eventually (actually now, since I want composite PKs for sync tests).

On the other hand, switching from defining the PK at the field level to defining it at the table-level does not seem so difficult in the future, given the number of tests for which we define a custom PK will probably be small.